pam/pam_test/pam-module-transaction-dummy_test.go

```diff
@@ -0,0 +1,218 @@
+package pam_test
+
+import (
+	"testing"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_ModuleTransactionDummy_SetItem(t *testing.T) {
+	t.Parallel()
+	tx := NewModuleTransactionDummy(nil)
+	err := tx.SetItem(pam.User, "an user")
+	require.NoError(t, err, "SetItem should not return an error")
+}
+
+func Test_ModuleTransactionDummy_GetItem_Empty(t *testing.T) {
+	t.Parallel()
+	tx := NewModuleTransactionDummy(nil)
+	val, err := tx.GetItem(pam.User)
+	require.NoError(t, err, "GetItem should not return an error")
+	require.Equal(t, "", val)
+}
+
+func Test_ModuleTransactionDummy_GetSetItem(t *testing.T) {
+	t.Parallel()
+	tx := NewModuleTransactionDummy(nil)
+	err := tx.SetItem(pam.User, "an user")
+	require.NoError(t, err, "SetItem should not return an error")
+	val, err := tx.GetItem(pam.User)
+	require.NoError(t, err, "GetItem should not return an error")
+	require.Equal(t, "an user", val)
+}
+
+func Test_ModuleTransactionDummy_GetEnv_Empty(t *testing.T) {
+	t.Parallel()
+	tx := NewModuleTransactionDummy(nil)
+	val := tx.GetEnv("AN_ENV")
+	require.Equal(t, "", val)
+}
+
+func Test_ModuleTransactionDummy_PutEnvGetEnv(t *testing.T) {
+	t.Parallel()
+	tx := NewModuleTransactionDummy(nil)
+	list, err := tx.GetEnvList()
+	require.NoError(t, err, "GetEnvList should not return an error")
+	require.Equal(t, map[string]string{}, list)
+
+	err = tx.PutEnv("AN_ENV")
+	require.NoError(t, err, "PutEnv should not return an error")
+
+	val := tx.GetEnv("AN_ENV")
+	require.Equal(t, "", val)
+
+	list, err = tx.GetEnvList()
+	require.NoError(t, err, "GetEnvList should not return an error")
+	require.Equal(t, map[string]string{}, list)
+
+	err = tx.PutEnv("OTHER_ENV=value")
+	require.NoError(t, err, "PutEnv should not return an error")
+	val = tx.GetEnv("OTHER_ENV")
+	require.Equal(t, "value", val)
+
+	list, err = tx.GetEnvList()
+	require.NoError(t, err, "GetEnvList should not return an error")
+	require.Equal(t, map[string]string{"OTHER_ENV": "value"}, list)
+
+	err = tx.PutEnv("OTHER_ENV=")
+	require.NoError(t, err, "PutEnv should not return an error")
+	val = tx.GetEnv("OTHER_ENV")
+	require.Equal(t, "", val)
+
+	list, err = tx.GetEnvList()
+	require.NoError(t, err, "GetEnvList should not return an error")
+	require.Equal(t, map[string]string{"OTHER_ENV": ""}, list)
+
+	err = tx.PutEnv("OTHER_ENV")
+	require.NoError(t, err, "PutEnv should not return an error")
+	val = tx.GetEnv("OTHER_ENV")
+	require.Equal(t, "", val)
+
+	list, err = tx.GetEnvList()
+	require.NoError(t, err, "GetEnvList should not return an error")
+	require.Equal(t, map[string]string{}, list)
+}
```

Comment:
Can we rewrite this using the `table testing` approach? The current implementation is a very C-like way of writing tests and, even though this is not wrong per se, in Go we tend to use table testing more. 
Using table testing would allow us to significantly reduce the number of test functions whilst also making it easier to understand the "contract" employed by the API. You can see some examples in the [broker_tests](https://github.com/ubuntu/authd/blob/main/internal/brokers/broker_test.go) and in the [cache_tests](https://github.com/ubuntu/authd/blob/main/internal/cache/db_test.go).

For example, this function would become something like `TestPutEnvGetEnv` and it would cover both success and error cases (I agree that it doesn't make much sense to use separate tests for `GetXX` functions, so it's fine to keep them together). Also, since the file is already named `pam-module-transaction-dummy_test.go`, you don't need to name the tests as `Test_ModuleTransactionDummy_XXX`, as the file name already points out to which source file the tests refer to, and, consequently, which functions we are testing (even though most of them are "methods", they appear a single time in the file, so no need to qualify the test name).

Reply:
> Also, please avoid resolving the comments yourself, if possible. The comments are very useful for the reviewer to navigate through what was requested and to see if they were addressed or not. 

Oh, fair enough... Normally in other projects I work on we use the "resolve" thing when it's an easy change to do to mark what comment has been addressed, but I can change that.

> If you're unsure how to move on with the table-testing approach, we can talk more about it in the specific comment, MM or HO.

No, as said already in another comment I'm well aware how to use it (see the https://github.com/msteinert/pam/pull/13 :-)), but it didn't look very useful here. But I can change that.

Reply:
I meant I wrote why I didn't use table-testing in https://github.com/ubuntu/authd/pull/89#discussion_r1365277370

Reply:
Yep, Gabriel submitted his review while I was writing mine, so sadly I didn't see it until it was too late :smiling_face_with_tear:. I'm sorry for the repetition.

Even though the tests are simple and this is not supposed to be used in production, I still think it's worth switching to table testing to stay consistent with the rest of the project.

Reply:
Ok, this is done now. :)